docs/testplan/platform/nvidia_new_thermal_control_algorithm_test_plan.md

```diff
@@ -0,0 +1,155 @@
+```
+
+
+### Supported topology
+The tests will be supported on any topo.
+
+
+### Test cases #1 - Check tc service behaviors after reboot
+1. Set hw-management-tc service to enabled state
+2. Reboot system( Randomly select cold,warm,fast, config reload)
+3. Wait for TC init (~120 sec after login prompt)
+4. Check syslog messages
+5. Check service status 
+
+### Test cases #2 - Check tc service behaviors when start and stop TC service
+1. Stop hw-management-tc service
+2. Check syslog and service state
+3. Check PWM value. It should be at maximum.
+4. Start hw-management-tc service.
+5. Immediately after start  - check PWM value . It should be at maximum
+6. wait 120 sec for svc init
+7. Check syslog and service state
+8. Check PWM value. It should be in range 20~35% (not maximum)
+
+### Test cases #3 - Check tc service behaviors when hw-management is not fully initialized.
+1. Stop hw-management-tc service
+2. Stop hw-management service
+3. Start hw-management-tc service
+4. Check PWM value. It should be at maximum
+5. Wait 3 minutes
+6. Check PWM value. It should be at maximum
+
+### Test cases #4 - Check PWM is the correct value when suspend and resume TC 
+1. Start TC svc.
+2. Check PWM in a low value like range 20~35% 
+3. Set TC suspend with echo 1 >> /var/run/hw-management/config/suspend & wait 30sec
+4. Check PWM is maximum (100%)
+5. Set TC resume with echo 0 >> /var/run/hw-management/config/suspend & wait 30sec
+6. Check PWM in a low value like range 20~35% 
+
+### Test cases #5 - Temperature sweep 
+1. Save the soft-links of tested sensors. Create static files with the same file name for the tested sensors.
+2. Get tested sensor parameters from /var/run/hw-management/config/tc_config.json like below e.g.:
+   {"asic":           {"pwm_min": 20, "pwm_max" : 100, "val_min":"!70000", "val_max":"!105000", "poll_time": 3}}
+3. Mock tested sensor temperature to val_min/10000-1. 
+   Wait poll_time secs
+4. Mock temperature increase
+   - Increase sensor temperature value with step 10
+   - Calculate the expected PWM with below formula:
+     - PWM = pwm_min + ((new_temp - val_min)/(val_max-val_min)) * (pwm_max - pwm_min)
+   - Wait poll_time secs, and get current system PWM by reading /var/run/hw-management/thermal/pwm1
+   - Check current_system_pwm >= expected_pwm
+   - Iterate the above steps until temperature is over val_max
+5. Mock temperature decrease
+   - decrease sensor temperature value with step 10
+   - Calculate the expected PWM with below formula:
+     - PWM = pwm_min + ((new_temp - val_min)/(val_max-val_min)) * (pwm_max - pwm_min)
+   - Wait poll_time secs, and get current system PWM
+   - Check current_system_pwm >= expected_pwm
+   - Iterate the above steps until temperature is over val_max
+6. Repeat step 1~ step 5  for following sensors or randomly select one sensor to test:
+   - asic
+   - cpu-core
+   - comex_voltmon1_temp_input
+   - voltmon1_temp_input
+   - module{X}
+   - ambient
+   - psu temp (1 ramdom PSU)
```

Comment:
gearbox should also be tested for tigris

Reply:
Done